### PR TITLE
Add feature toggles to docs

### PR DESCRIPTION
**Pull request by mjseaman**
**What is this feature?**

Customers use feature toggles frequently, but they lack a reference to which toggles are available to them. This adds a reference, so customers can choose which features to experiment with.

**Who is this feature for?**

Users and customer admins who want to experiment with new features.

**Note**

This is a draft PR, pending the more significant work to generate the table of feature toggles based on https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go

**Docs generation notes for @ryantxu **

- I like feature toggle name in code brackets, implying users should literally use that text for their feature toggles
- I'd love to see a link to the docs for each feature, where available. Ideas: 
  - Description is overrideable, so that we can editorialize or add links to applicable docs
  - Add an extra column like Additional Notes where we could add those details on this docs page 
  - Add a docs path field to registry.go and then pull that into the doc here. 
- There's an open discussion about feature release stages; the current candidate names are Experimental, Beta, and Generally Available. I'm thinking we can map it like this:
  - FeatureStateAlpha = Experimental
  - FeatureStateBeta = Beta
  - FeatureStateStable = Generally Available
- Is there any way to *disable* a feature toggle that is `Expression:   "true"`? If so, then we should include that as a new column called "Enabled by default?" and if not, we should just filter out those toggles from this doc.
- Is there a way to tell when a given feature toggle was added? It would be nice to tell old ones apart from new ones.
- Since features might move from one release stage to another, I think it would be best to sort feature toggles alphabetically by name. This will make them easier to find and refer back to. However I don't have a strong opinion about this..

### CODE REVIEW

**Comment by mjseaman** [comment]
@chri2547 FYI! I placed this doc inside the Configure Grafana section but would be totally up for moving it around. I also wonder if I placed and named it correctly - on my local instance the docs page didn't actually render..

**Comment by ryantxu** [comment]
> Is there a way to tell when a given feature toggle was added? It would be nice to tell old ones apart from new ones.

From my IDE, it is easy, but not from our docs.  In the IDE, you can click on a line and see who edited it when :) git is pretty amazing!

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+description: Learn about toggles for experimental and beta features, which you can enable or disable.
+title: Configure feature toggles
+weight: 150
+---
+
+# Configure feature toggles
+
+Feature toggles, also known as feature flags, are used for experimental or beta features in Grafana. Although we do not recommend that you use these features in production, you can turn on feature toggles to try out new functionality in development or test environments.
+
+This page contains a list of available feature toggles. To learn how to turn on feature toggles, refer to our [Configure Grafana documentation]({{< relref "../_index.md/#feature_toggles" >}}). Feature toogles are also available to Grafana Cloud Advanced customers - if you use Grafana Cloud Advanced, you can open a support ticket specifying the feature toggles and stack you would like them enabled
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:13]: I would add the instructions for turning on a feature toggle here. It duplicates content, but a very small amount. It'll provide a nicer user experience not to navigate to another set of docs.

Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:13]:
```suggestion
This page contains a list of available feature toggles. To learn how to turn on feature toggles, refer to our [Configure Grafana documentation]({{< relref "../_index.md/#feature_toggles" >}}). Feature toggles are also available to Grafana Cloud Advanced customers. If you use Grafana Cloud Advanced, you can open a support ticket and specify the feature toggles and stack for which you want them enabled.
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+## Stable feature toggles
+
+Some stable features are enabled by default -- they can be disabled by setting the flag to false in the configuration.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:17]:
```suggestion
Some stable features are enabled by default. You can disable a stable feature by setting the feature flag to `false` in the configuration.
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+| Feature toggle name          | Description                                                                                                     | Enable by default |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:19]:
```suggestion
| Feature toggle name          | Description                                                                                                     | Enabled by default |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+| ---------------------------- | --------------------------------------------------------------------------------------------------------------- | ----------------- |
+| `promQueryBuilder`           | Show prometheus query builder                                                                                   | Yes               |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:21]:
```suggestion
| `promQueryBuilder`           | Show Prometheus query builder                                                                                   | Yes               |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+| `disableEnvelopeEncryption`  | Disable envelope encryption (emergency only)                                                                    |                   |
+| `database_metrics`           | Add prometheus metrics for database tables                                                                      |                   |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:23]:
```suggestion
| `database_metrics`           | Add Prometheus metrics for database tables                                                                      |                   |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+## Alpha feature toggles
+
+These are features early in their development lifecycle, they are not yet supported in grafana cloud. Alpha features may
+be changed or removed without notice.
+
+| Feature toggle name                | Description                                                                                                                                                 |
+| ---------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `alertingBigTransactions`          | Use big transactions for alerting database writes                                                                                                           |
+| `dashboardPreviews`                | Create and show thumbnails for dashboard search results                                                                                                     |
+| `live-config`                      | Save grafana live configuration in SQL tables                                                                                                               |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:56]:
```suggestion
| `live-config`                      | Save Grafana Live configuration in SQL tables                                                                                                               |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+| `live-pipeline`                    | enable a generic live processing pipeline                                                                                                                   |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:57]:
```suggestion
| `live-pipeline`                    | Enable a generic live processing pipeline                                                                                                                   |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+| `live-service-web-worker`          | This will use a webworker thread to processes events rather than the main thread                                                                            |
+| `queryOverLive`                    | Use grafana live websocket to execute backend queries                                                                                                       |
+| `tempoApmTable`                    | Show APM table                                                                                                                                              |
+| `influxdbBackendMigration`         | Query InfluxDB InfluxQL without the proxy                                                                                                                   |
+| `publicDashboards`                 | enables public access to dashboards                                                                                                                         |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:62]:
```suggestion
| `publicDashboards`                 | Enables public access to dashboards                                                                                                                         |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+| `lokiLive`                         | support websocket streaming for loki (early prototype)                                                                                                      |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:63]:
```suggestion
| `lokiLive`                         | Support WebSocket streaming for Loki (early prototype)                                                                                                      |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+| `lokiDataframeApi`                 | use experimental loki api for websocket streaming (early prototype)                                                                                         |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:64]:
```suggestion
| `lokiDataframeApi`                 | Use experimental Loki API for WebSocket streaming (early prototype)                                                                                         |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+| `redshiftAsyncQueryDataSupport`    | Enable async query data support for Redshift                                                                                                                |
+| `athenaAsyncQueryDataSupport`      | Enable async query data support for Athena                                                                                                                  |
+| `increaseInMemDatabaseQueryCache`  | Enable more in memory caching for database queries                                                                                                          |
+| `newPanelChromeUI`                 | Show updated look and feel of grafana-ui PanelChrome: panel header, icons, and menu                                                                         |
+| `showDashboardValidationWarnings`  | Show warnings when Dashboards do not validate against the schema                                                                                            |
+| `mysqlAnsiQuotes`                  | Use double quote to escape keyword in Mysql query                                                                                                           |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:88]:
```suggestion
| `mysqlAnsiQuotes`                  | Use double quotes to escape keyword in a MySQL query                                                                                                           |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+| `elasticsearchBackendMigration`    | Use Elasticsearch as backend data source                                                                                                                    |
+| `secureSocksDatasourceProxy`       | Enable secure socks tunneling for supported core datasources                                                                                                |
+| `authnService`                     | Use new auth service to perform authentication                                                                                                              |
+
+## Development feature toggles
+
+The following toggles require explicitly setting Grafana's [app mode]({{< relref "../_index.md/#app_mode" >}}) to 'development' before you can enable this feature toggle. These features tend to be especially experimental.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:95]:
```suggestion
The following toggles require explicitly setting Grafana's [app mode]({{< relref "../_index.md/#app_mode" >}}) to 'development' before you can enable this feature toggle. These features tend to be experimental.
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+| Feature toggle name                    | Description                                               |
+| -------------------------------------- | --------------------------------------------------------- |
+| `dashboardPreviewsAdmin`               | Manage the dashboard previews crawler process from the UI |
+| `showFeatureFlagsInUI`                 | Show feature flags in the settings UI                     |
+| `dashboardsFromStorage`                | Load dashboards from the generic storage interface        |
+| `export`                               | Export grafana instance (to git, etc)                     |
+| `azureMonitorResourcePickerForMetrics` | New UI for Azure Monitor Metrics Query                    |
+| `grpcServer`                           | Run GRPC server                                           |
+| `objectStore`                          | SQL based object store                                    |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:105]:
```suggestion
| `objectStore`                          | SQL-based object store                                    |
```

**Review by chri2547 — COMMENTED** [review]
Added suggestions throughout. Thanks for working on this!

**Review by ryantxu — APPROVED** [review]
LGTM -- we need to get everyone to audit their feature flags before 9.4

**Pull request merged**